### PR TITLE
Add a REST api for configuring aci-gw and pull in contivmodel changes

### PR DESCRIPTION
**Pull request by jojimt**

### CODE REVIEW

File: vendor/github.com/contiv/contivmodel/bgphost.json [hunk 1]
```diff
@@ -15,7 +15,6 @@
             "routerip": {
                 "type": "string",
                 "title": "Bgp router intf ip",
-                "length": 15,
```
Comment (gaurav-dalvi) [inline vendor/github.com/contiv/contivmodel/bgphost.json]: why this change in this PR ?

File: netctl/commands.go [hunk 2]
```diff
@@ -450,6 +450,56 @@ var Commands = []cli.Command{
...
+					cli.StringFlag{
+						Name:  "node-bindings, n",
+						Usage: "Blank or  Comma separated nodes of the form topology/pod-1/node-101",
+					},
+					cli.StringFlag{
+						Name:  "phys-dom, d",
+						Usage: "ACI physical domain name (e.g. containerDom)",
+					},
+					cli.StringFlag{
+						Name:  "enforce-policies, e",
```
Comment (gaurav-dalvi) [inline netctl/commands.go]: regarding this and option below : what is the purpose of these ones ?

Comment (jojimt) [inline netctl/commands.go]: This is one of the existing env vars. The second one is to utilize the common tenant in aci.

Comment (jojimt) [inline netctl/commands.go]: Correct.

File: netctl/netctl.go [hunk 3]
```diff
@@ -904,6 +904,65 @@ func setGlobal(ctx *cli.Context) {
...
+			writer.Write([]byte(fmt.Sprintf("Path bindings: %v\n", gw.PathBindings)))
+			writer.Write([]byte(fmt.Sprintf("Node bindings: %v\n", gw.NodeBindings)))
+			writer.Write([]byte(fmt.Sprintf("Physical domain: %v\n", gw.PhysicalDomain)))
+			writer.Write([]byte(fmt.Sprintf("Enforce policies: %v\n", gw.EnforcePolicies)))
+			writer.Write([]byte(fmt.Sprintf("Include Common-Tenant: %v\n", gw.IncludeCommonTenant)))
+		}
+	}
+}
+
+func inspectAciGw(ctx *cli.Context) {
```
Comment (gaurav-dalvi) [inline netctl/netctl.go]: Could you please write aci-inspect test case ? We have all inspect test cases here : https://github.com/contiv/netplugin/tree/master/test/integration

File: netmaster/objApi/objapi_test.go [hunk 4]
```diff
@@ -309,6 +309,45 @@ func checkGlobalSet(t *testing.T, expError bool, fabMode, vlans, vxlans, fwdMode
 	}
 }
 
+// checkAciGwSet sets AciGw state and verifies verifies it
+func checkAciGwSet(t *testing.T, expError bool, enf, comTen, paths, nodes, domain string) {
```
Comment (gaurav-dalvi) [inline netmaster/objApi/objapi_test.go]: nit pick : 
comTen -> commonTenant
enf -> enforcePolicy

Comment (jojimt) [inline netmaster/objApi/objapi_test.go]: Please don't nit pick variable names in test code.

File: netctl/netctl.go (hunk 3 above)
Comment (jojimt) [inline netctl/netctl.go]: Yes, will be there when I add gw interface (i.e. part2)

File: netmaster/objApi/objapi_test.go [hunk 5]
```diff
@@ -1239,6 +1278,31 @@ func TestGlobalSetting(t *testing.T) {
...
+	// basic create and delete
+	checkAciGwSet(t, false, "yes", "yes", "topology/pod-1/paths-101/pathep-[eth1/14]", "", "testDom")
+	checkAciGwDelete(t, false, "aciGw")
+
+	checkAciGwSet(t, true, "yes", "yes", "pod-1/paths-101/pathep-[eth1/14]", "", "testDom")
+	checkAciGwSet(t, false, "yes", "no", "topology/pod-1/paths-101/pathep-[eth1/14]", "", "testDom")
+	checkAciGwSet(t, true, "yes", "no", "topology/pod-1/paths-101/pathep-[eth1/14],pod-1/paths-101/pathep-[eth1/15]", "", "testDom")
+	checkAciGwSet(t, false, "yes", "no", "topology/pod-1/paths-101/pathep-[eth1/14],topology/pod-1/paths-101/pathep-[eth1/15]", "", "testDom")
+	checkAciGwSet(t, true, "yes", "yes", "topology/pod-1/paths-101/pathep-[eth1/14],topology/pod-1/paths-101/pathep-[eth1/15]", "topology/tor-1/node-101", "testDom")
+	checkAciGwSet(t, false, "yes", "yes", "topology/pod-1/paths-101/pathep-[eth1/14],topology/pod-1/paths-101/pathep-[eth1/15]", "topology/pod-1/node-101", "testDom")
```
Comment (gaurav-dalvi) [inline netmaster/objApi/objapi_test.go]: Can you add mix of Path and Node positive test case ?

**Comment by gaurav-dalvi** [comment]
LGTM !
